google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/ValidateTextAd.java

```diff
@@ -0,0 +1,178 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.ads.googleads.examples.campaignmanagement;
+
+import com.beust.jcommander.Parameter;
+import com.google.ads.googleads.examples.utils.ArgumentNames;
+import com.google.ads.googleads.examples.utils.CodeSampleParams;
+import com.google.ads.googleads.lib.GoogleAdsClient;
+import com.google.ads.googleads.v2.common.ExpandedTextAdInfo;
+import com.google.ads.googleads.v2.common.PolicyTopicEntry;
+import com.google.ads.googleads.v2.enums.AdGroupAdStatusEnum.AdGroupAdStatus;
+import com.google.ads.googleads.v2.errors.GoogleAdsError;
+import com.google.ads.googleads.v2.errors.GoogleAdsException;
+import com.google.ads.googleads.v2.errors.PolicyFindingErrorEnum.PolicyFindingError;
+import com.google.ads.googleads.v2.resources.Ad;
+import com.google.ads.googleads.v2.resources.AdGroupAd;
+import com.google.ads.googleads.v2.services.AdGroupAdOperation;
+import com.google.ads.googleads.v2.services.AdGroupAdServiceClient;
+import com.google.ads.googleads.v2.services.MutateAdGroupAdsResponse;
+import com.google.ads.googleads.v2.utils.ResourceNames;
+import com.google.common.collect.ImmutableList;
+import com.google.protobuf.StringValue;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Shows how to use the validateOnly header to validate an expanded text ad. No objects will be
```

Comment:
I think "no ads entities will be created" is a little more accurate here, there will technically be several objects created in the process, the changes won't be persisted though.